_posts/2020-11-04-differences-between-programming-languages.md

```diff
@@ -0,0 +1,158 @@
+
+## 4. Bash
+![Bash](https://www.fullstackpython.com/img/logos/bash-wide.jpg)
+
+Bash is a scripting language. It is a a command processor and it is a Unix shell. Bash can run in a text editor window where user types commands and runs to make actions. Bash can also execute commands from a shell script file.
+
+Bash was originally released in 1989 written by Brian Fox. The language was designed primarily as a free software replacement for the Bourne shell. [^5]
+
+Bash contains a variety of features, including wildcard matching, piping, control structures, command history and brace expansion, etc. It has both features from the C shell's major advantages and its own features. [^6]
+
+### Bash Script Example
+```
+#!/bin/bash
+n=0
+if [ $n -eq 0 ];
+then
+echo "It is number 0"
+else
+echo "It is not number 0"
+fi
+```
+Run the file in the bash command line:
+```
+$ bash if_example.sh
+```
+
+Some common bash commands: `ls`, `echo`, `touch`,`mkdir`,`pwd`, `grep`, `cd`, `mv`,  `|`- Pipe, `head`, `clean`...
+
+## 5. HTML
+![HTML](https://upload.wikimedia.org/wikipedia/commons/thumb/6/61/HTML5_logo_and_wordmark.svg/240px-HTML5_logo_and_wordmark.svg.png)
+
+HTML, Hypertext Markup Language, is a markup language designed for the development of web browser. You can insert and edit text, images, videos and other media via HTML. It aims to provide structured formatting so that users can clearly see what developers want to display in the webpage. Almost everything we use and see on the website is fundamentally coded in HTML. If you want to make your own website, HTML is absolutely a nice choice. Even for the complex website formatting style, HTML can still handle it.
+
+HTML is assisted by CSS (Cascading Style Sheets) and JavaScript. If you want to learn HTML, be prepared to learn CSS and JavaScript as well.  
+
+### HTML History
+HTML was developed my a physicist Tim Berners-Lee. His purpose of developing this system was to share documents. In the late 1980, he wrote the browser and server software. The first version was published in November 1995 in order to upload file and client-side image. In the latest version HTML 5, it was published as a W3C Recommendation. [^7]
+
+![Tim Berners](http://archive.thegovlab.org/wp-content/uploads/2014/01/Sir-Tim-Berners-Lee.png)
+
+# Example of HTML
+```
+<!-- This is a comment -->
+<html>
+  <head>
+    <title>This is a title</title>
+  </head>
+  <body>
+    <div>
+        <p>Hello world!</p>
+    </div>
+  </body>
+</html>
+```
```

Comment:
@DingShen29 Similar to my 2 previous comments, a brief explanation of this code might be helpful.